lib/driver.js

```diff
@@ -182,15 +185,38 @@ class YouiEngineDriver extends BaseDriver {
     let iosArgs = {
       javascriptEnabled: true,
     };
-    let iosdriver = new IOSDriver(iosArgs);
+    let driverName = "IOSDriver";
+    if (opts.platformVersion) {
+      let majorVer = opts.platformVersion.toString().split(".")[0];
+      if (parseInt(majorVer, 10) >= 10) {
+        driverName = "XCUITestDriver";
+      }
+    }
+
+    let iosdriver;
+    if (driverName === "XCUITestDriver") {
+      logger.info("Selected driver: XCUITestDriver");
+      iosdriver = new XCUITestDriver(iosArgs);
+    } else {
+      logger.info("Selected driver: IOSDriver");
+      iosdriver = new IOSDriver(iosArgs);
+    }
```

Comment:
If I move it above, I have 2 else scenarios where I need would trigger IOSDriver. I had it like that before and it looked messy.
If platform version isn't supplied, we run IOSDriver